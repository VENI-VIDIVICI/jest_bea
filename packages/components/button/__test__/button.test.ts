import { mount } from "@vue/test-utils"
import { Button } from "../index"
import { h } from "vue"
describe("sum", () => {
  it("button:click", () => {
    const wrap = mount(Button, {
      slots: {
        default: "按钮"
      }
    })
    wrap.find("button").trigger("click")
    expect(wrap.emitted("click")).toBeTruthy()
  });
  it("button:disabled", () => {
    const wrap = mount(Button, {
      props: {
        disabled: true
      }
    })
    wrap.find("button").trigger("click")
    expect(wrap.emitted("click")).toBeFalsy()
  });
  it("button:slots", () => {
    const wrap = mount(Button, {
      slots: {
        default: h("div", "按钮")
      }
    })
    expect(wrap.text()).toBe("按钮")
    expect(wrap.html()).toMatchSnapshot()
  });
});
