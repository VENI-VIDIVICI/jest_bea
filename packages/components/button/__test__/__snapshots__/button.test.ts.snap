// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sum button:slots 1`] = `
"<button class="btn t-button--theme-primary">
  <div>按钮</div>
</button>"
`;
